pandas/core/dtypes/common.py

```diff
@@ -1219,7 +1219,18 @@ def is_bool_dtype(arr_or_dtype) -> bool:
 
     if isinstance(arr_or_dtype, ABCIndex):
         # Allow Index[object] that is all-bools or Index["boolean"]
-        return arr_or_dtype.inferred_type == "boolean"
+        if arr_or_dtype.inferred_type == "boolean":
```

Comment:
Hmm this looks unrelated. Did a merge conflict go wrong in this PR?

Reply:
agreed, will revert

Reply:
Updated + green